ITISFoundation/osparc-simcore
# 🐛⚗️ Add CI test to check for git commit / docker container SHA missmatch …

## What do these changes do?

On Dec01-2021 we noticed that in some cases the deployment workflow, where docker images of the services are built in the github-actions CI workflows, will not work correctly without error. In this case, the container label and code does not match the github-commit-code it should be associated to. 
This is a first version of a script which checks for these possible missmatches in order to catch the errors.


## Related issue/s

No issue filed for this problem so far.

## How to test

I tested locally using `act` to mock a github-actions run. This seemed fine. Mimicking a github actions run properly is a pain though, so ideally we want to see what happens with this test on github-actions properly


## Checklist
- [ ] Should we only print errors and continue for now to not break workflows or actually return non-zero and thus make the entire deployment pipeline fail? The latter in the long run of course, but what is the short-term immediate strategy?
- [ ] The actual comparison could maybe be more rebust, as currently it is only checking is a substring is in a string.
- [ ] More comments to explain what is going on (and why)
